server/src/repositories/Store/poRepository.ts

```diff
@@ -0,0 +1,187 @@
+import { knex } from "../../db";
+import { PurchaseOrderMappper } from "../../mappers/store/PurchaseOrderMapper";
+import { InventoryItem } from "../../models/store/inventoryItem";
+import { PurchaseOrder } from "../../models/store/purchaseOrder";
+import { PurchaseOrderItemInput } from "../../models/store/purchaseOrderItemInput";
+
+export interface IPurchaseOrderRepo {
+  getAllPOs(): Promise<PurchaseOrder[]>;
+  getPOById(poId: number): Promise<PurchaseOrder>;
+  createNewPO(creatorId: number, expectedDeliveryDate: Date, items: PurchaseOrderItemInput[], attachments: string[]): Promise<PurchaseOrder>;
+  addItemToPO(item: InventoryItem, poId: number, count: number): Promise<PurchaseOrder>;
+  addItemsToPO(poId: number, items: PurchaseOrderItemInput[]): Promise<PurchaseOrder>;
+  addAttachmentToPO(attachment: string, poId: number): Promise<PurchaseOrder>;
+  addAttachmentsToPO(attachments: string[], poId: number): Promise<PurchaseOrder>;
+  removeAttachmentFromPO(attachment: string, poId: number): Promise<PurchaseOrder>;
+  removeItemFromPO(itemId: number, poId: number): Promise<PurchaseOrder>;
+  updateItemCountInPO(itemId: number, count: number, poId: number): Promise<PurchaseOrder>;
+  deletePOById(poId: number): Promise<void>;
+}
+
+
+export class PurchaseOrderRepo implements IPurchaseOrderRepo {
```

Comment:
Also can we rename this file to `purchaseOrderRepository` instead of `poRepository`